File: dotnet/Library/DIDComm.Messaging.gRPC/DIDKey.cs
namespace DIDComm.Messaging
{
    public class DIDKey
    {
        /// <summary>
        /// Generate new key
        /// </summary>
        /// <param name="request"></param>
        /// <returns></returns>
        public static GenerateKeyResponse Generate(GenerateKeyRequest request) =>
            DIDComm.Call<GenerateKeyRequest, GenerateKeyResponse>(request, NativeMethods.didkey_generate);

        public static ResolveResponse Resolve(ResolveRequest request) =>
            DIDComm.Call<ResolveRequest, ResolveResponse>(request, NativeMethods.didkey_resolve);
    }
}

File: java/src/DIDComm/Messaging/gRPC/DIDKey.java
package DIDComm.Messaging.gRPC;

import DIDComm.Messaging.Proto.API;
import com.google.protobuf.InvalidProtocolBufferException;

public class DIDKey {
    public static API.GenerateKeyResponse Generate(API.GenerateKeyRequest request) throws InvalidProtocolBufferException {
        return API.GenerateKeyResponse.parseFrom(NativeMethods.didkey_generate(request.toByteArray()));
    }

    public static API.ResolveResponse Resolve(API.ResolveRequest request) throws InvalidProtocolBufferException {
        return API.ResolveResponse.parseFrom(NativeMethods.didkey_resolve(request.toByteArray()));
    }
}
